src/main/java/es/unizar/smartcampuz/application/controller/DashboardController.java

```diff
@@ -115,43 +124,30 @@
 
     @GetMapping("/availableHours")
     @ResponseBody
-    public ResponseEntity<String> listAvailableHours (HttpServletRequest request) throws IOException{
-        String location = request.getHeader("location");
-        int day = Integer.parseInt(request.getHeader("day"));
-        int month = Integer.parseInt(request.getHeader("month"));
+    public ResponseEntity<String> listAvailableHours(@RequestParam String location, @RequestParam int day,
+                                                      @RequestParam int month) throws IOException, ParseException{
 
         //TODO: ¿Comprobar que la localización existe?
 
-        if(!isValidDate(day, month)){
-            return new ResponseEntity<>("\"La fecha no es válida\"", HttpStatus.BAD_REQUEST);
-        }
+//        if(!isValidDate(day, month)){
+//            return new ResponseEntity<>("\"La fecha no es válida\"", HttpStatus.BAD_REQUEST);
+//        }
 
-        //TODO: Pedir reservas aprobadas de una localización en un día concreto
-        Iterable<?> iter = new ArrayList();
+        Set<Reservation> approvedReservations = reservationRepository.findAllByRoomIDAndDateAndState(
+            location, dateFormater.parse(String.format("%02d%02d2017", day, month)), ReservationState.APPROVED);
         // Initializes the array with 'false' value in all it's fields
         boolean [] availableHours = new boolean[24];
+        Arrays.fill(availableHours, true);
```

Comment:
That sounds like a good idea. I will replace the `Arrays.fill(availableHours, true);` with a little loop that only puts `true` in the actual available hours.